lib/png/generate_png.js

```
@@ -0,0 +1,22 @@
+async (imgString) => {
+  const tempImg = new Image();
+  const loadImage = () => {
+    return new Promise((resolve, reject) => {
+      tempImg.onload = (event) => resolve(event.currentTarget);
+      tempImg.onerror = () => {
+        reject("error loading string as an image");
+      };
+      tempImg.src = imgString;
+    });
+  };
+  const img = await loadImage();
+  const canvas = document.createElement("canvas");
+  canvas.width = img.width;
+  canvas.height = img.height;
+  const ctx = canvas.getContext("2d");
+  if (!ctx) {
+    return "";
```

reject?

---

Why reject, there's no promise here

---

we can return an error instead